R/object_usage_linter.R

```diff
@@ -166,63 +168,46 @@ extract_glued_symbols <- function(expr) {
   #
   # Package stringr:
   #  - str_interp
-  glue_calls <- xml_find_all(
-    expr,
-    xpath = paste0(
-      "descendant::SYMBOL_FUNCTION_CALL[text() = 'glue']/", # a glue() call
-      "parent::expr[",
-      # without .envir or .transform arguments
-      "not(following-sibling::SYMBOL_SUB[text() = '.envir' or text() = '.transform']) and",
-      # argument that is not a string constant
-      "not(following-sibling::expr[not(STR_CONST)])",
-      "]/",
-      # get the complete call
-      "parent::expr"
-    )
-  )
-
-  if (length(glue_calls) == 0L) {
-    return(character())
-  }
+  glue_call_xpath <- "
+    descendant::SYMBOL_FUNCTION_CALL[text() = 'glue']
+      /parent::expr
+      /parent::expr[
+        not(SYMBOL_SUB[text() = '.envir' or text() = '.transform'])
+        and not(expr[position() > 1 and not(STR_CONST)])
```

Comment:
Why `position() > 1`?

Reply:
`position()=1` is the `SYMBOL_FUNCTION_CALL` `<expr>`

Reply:
Ah now I see you moved one expr further up in comparison to the old XPath.

Reply:
added a comment

Reply:
right -- saves some `following-sibling::` space, especially since the end anchor of the XPath is that `expr` anyway.